examples/platforms/kw41z/radio.c

```diff
@@ -234,24 +234,25 @@ otError otPlatRadioReceive(otInstance *aInstance, uint8_t aChannel)
 
     sState = OT_RADIO_STATE_RECEIVE;
 
-    otEXPECT(rf_get_state() != XCVR_RX_c);
-
-    rf_abort();
-
-    /* Set Power level for auto TX */
-    rf_set_tx_power(sAutoTxPwrLevel);
-    rf_set_channel(aChannel);
-    sRxFrame.mChannel = aChannel;
+    /* Check if the channel needs to be changed */
+    if (sChannel != aChannel)
```

Comment:
If the first call to otPlatRadioReceive() is done with the aChannel parameter's value equal to the DEFAULT_CHANNEL, the radio will not start the RX sequence. I suggest you to modify the if statement to something like this: if ((sChannel != aChannel) || (rf_get_state() != XCVR_RX_c)) or if ((sChannel != aChannel) || (sState != OT_RADIO_STATE_RECEIVE))